### ISSUE TITLE
Early Stopping の追加

### ISSUE THREAD

**Issue by kurorosu** [kurorosu/pochitrain#117]
## 概要

現在, 学習は指定エポック数まで必ず実行される. 検証精度が改善しなくなった場合に自動で学習を停止する Early Stopping 機能を追加し, 過学習の防止と学習時間の短縮を図りたい.

## 現状

- `PochiTrainer` は `num_epochs` で指定されたエポック数を全て実行する
- ベストモデルの保存は `best_accuracy` の更新時に行われている（`pochi_trainer.py`）
- 学習の途中停止は手動の Ctrl+C のみ

### 1. Early Stopping の設定を config に追加

```python
# configs/pochi_train_config.py
early_stopping = {
    "enabled": True,
    "patience": 10,          # 改善なしで許容するエポック数
    "min_delta": 0.0,        # 改善とみなす最小変化量
    "monitor": "val_accuracy",  # 監視する指標
}
```

### 2. Early Stopping ロジックの実装

- `PochiTrainer` の学習ループ内で, 検証精度が `patience` エポック連続で `min_delta` 以上改善しなかった場合に学習を停止
- 停止時にログ出力（停止理由, 最終エポック, ベスト精度）
- `early_stopping` 設定が未指定または `enabled: False` の場合は従来通り全エポック実行

### 3. 設計方針

- SRP に従い, Early Stopping の判定ロジックは独立したクラスとして実装する
- `PochiTrainer` は Early Stopping クラスのインスタンスを保持し, エポック終了時に判定を委譲する

## 対象ファイル

- `pochitrain/early_stopping.py` — Early Stopping クラスの新規作成
- `pochitrain/pochi_trainer.py` — 学習ループへの Early Stopping 統合
- `pochitrain/validation/` — config バリデーションの追加
- `configs/pochi_train_config.py` — 設定例の追加
- `tests/` — 対応するテストの追加

## Labels

`enhancement`

## Branch

`feature/early-stopping`
